carbonmark-api/src/routes/purchases/[id]/get.ts

```diff
@@ -1,51 +1,81 @@
-import { FastifyInstance } from "fastify";
-import { isNil } from "lodash";
-import { GetPurchasesByIdQuery } from "../../../.generated/types/marketplace.types";
+import { Static } from "@sinclair/typebox";
+import { isHexString } from "ethers-v6";
+import { FastifyInstance, FastifyReply, FastifyRequest } from "fastify";
+import { Purchase as MumbaiPurchase } from "../../../graphql/marketplaceMumbai.types";
+import { NetworkParam } from "../../../models/NetworkParam.model";
 import { Purchase } from "../../../models/Purchase.model";
+import { CreditId } from "../../../utils/CreditId";
 import { gqlSdk } from "../../../utils/gqlSdk";
-import { ParamsT, schema } from "./get.schema";
+import { fetchCarbonProject } from "../../../utils/helpers/carbonProjects.utils";
+import { schema } from "./get.schema";
 
-const routeHandler = (fastify: FastifyInstance) =>
-  fastify.route<{
-    Params: ParamsT;
-    Reply: Purchase | { error: string };
-  }>({
-    method: "GET",
-    url: "/purchases/:id",
-    handler: async (request, reply) => {
-      let response: GetPurchasesByIdQuery;
-      try {
-        response = await gqlSdk.marketplace.getPurchasesById(request.params);
-      } catch (error) {
-        // Return bad gateway and pass the error
-        console.error(error);
-        return reply.status(502).send(error?.message);
-      }
+/** Purchase ids are a txn hash */
+const isValidPurchaseId = (id?: string | null) => {
+  if (!id) return false;
+  return id.length === 66 && isHexString(id);
+};
 
-      const data = response.purchases?.at(0);
-      /** Handle the not found case */
-      if (isNil(data)) {
-        return reply.status(404).send({ error: "Purchase not found" });
-      }
+const getPurchaseById = async (params: {
+  id: string | null;
+  network?: NetworkParam;
+}): Promise<Purchase | MumbaiPurchase | null> => {
+  const graph =
+    params.network === "mumbai" ? gqlSdk.marketplaceMumbai : gqlSdk.marketplace;
+  const response = await graph.getPurchasesById({
+    id: params.id,
+  });
+  return response.purchases?.at(0) || null;
+};
```

Comment:
No biggie but can we put these in a `.util`? even `purchases/[id]/get.util.ts`

Reply:
Will do, to follow the conventions you've established in `carbonmark-api` 👍 
Got me curious though, so I posted in dev chit chat for some more opinions on code organization